internal/models/hype_train.go

```diff
@@ -0,0 +1,91 @@
+	TopContributions 			[]ContributionData  	`json:"top_contributions"`
+	Total      				    int64 				 	`json:"total"`
+}
+
+type ContributionData struct{
+	TotalContribution   		 	 int64 	`json:"total"`
+	TypeOfContribution        		 string `json:"type"`
+	UserWhoMadeContribution	    	 string `json:"user_id"`
+	UserNameWhoMadeContribution	     string `json:"user_name"`
+	UserLoginWhoMadeContribution     string `json:"user_login"`
+}
+
+type HypeTrainWebSubResponse struct {
+	Data []HypeTrainWebSubEvent `json:"data"`
+}
+
+type HypeTrainEventBeginSubResponse struct {
+	Subscription EventsubSubscription         	`json:"subscription"`
+	Event        HypeTrainEventBeginSubEvent 	`json:"event"`
+}
+
+type HypeTrainEventProgressSubResponse struct {
+	Subscription EventsubSubscription         	`json:"subscription"`
+	Event        HypeTrainEventProgressSubEvent `json:"event"`
+}
+
+type HypeTrainEventEndSubResponse struct {
+	Subscription EventsubSubscription         	`json:"subscription"`
+	Event        HypeTrainEventEndSubEvent 		`json:"event"`
+}
+
+type HypeTrainEventBeginSubEvent struct {
+	BroadcasterUserID    		string 					`json:"broadcaster_user_id"`
+	BroadcasterUserLogin 		string 					`json:"broadcaster_user_login"`
+	BroadcasterUserName  		string 					`json:"broadcaster_user_name"`
+	Total   			 		int64 					`json:"total"`
+	Progress		     		int64 					`json:"progress"`
+	Goal       			 		int64					`json:"goal"`
+	TopContributions 	 		[]ContributionData  	`json:"top_contributions"`
+	LastContribution 			ContributionData 		`json:"last_contribution"`
+	StartedAtTimestamp		    string 					`json:"started_at"`
+	ExpiresAtTimestamp	        string 					`json:"expires_at"`
+}
+
+type HypeTrainEventProgressSubEvent struct {
+	BroadcasterUserID    		string 					`json:"broadcaster_user_id"`
+	BroadcasterUserLogin 		string 					`json:"broadcaster_user_login"`
+	BroadcasterUserName  		string 					`json:"broadcaster_user_name"`
+	Level   			 		int64 					`json:"level"`
+	Total   			 		int64 					`json:"total"`
+	Progress		     		int64 					`json:"progress"`
+	Goal       			 		int64					`json:"goal"`
+	TopContributions 	 		[]ContributionData  	`json:"top_contributions"`
+	LastContribution 			ContributionData 		`json:"last_contribution"`
+	StartedAtTimestamp		    string 					`json:"started_at"`
+	ExpiresAtTimestamp	        string 					`json:"expires_at"`
+}
+
+type HypeTrainEventEndSubEvent struct {
+	BroadcasterUserID    		string 					`json:"broadcaster_user_id"`
+	BroadcasterUserLogin 		string 					`json:"broadcaster_user_login"`
+	BroadcasterUserName  		string 					`json:"broadcaster_user_name"`
+	Level   			 		int64 					`json:"level"`
+	Total   			 		int64 					`json:"total"`
+	TopContributions 	 		[]ContributionData  	`json:"top_contributions"`
+	StartedAtTimestamp		    string 					`json:"started_at"`
+	EndedAtTimestamp	        string 					`json:"ended_at"`
+	CooldownEndsAtTimestamp	    string 					`json:"cooldown_ends_at"`
+}
```

Comment:
The names should be `...EventSubEvent` and `EventSubResponse`; just is the same as others.